### DISCUSSION TITLE
AdonisJS 5 + Prisma + MongoDB + Authentication

### DISCUSSION THREAD

**Discussion by PHDiniiz** [adonisjs/core#4249] [Help]
Is there any way to use Adonis with Prisma and perform the login and registration process via MongoDB? I watched the [AdonisJS Authentication with Prisma](https://adonismastery.com/watch/adonisjs-authentication-with-prisma) video where it shows the integration of Prisma with MongoDB, however, Redis is used instead of Database. I would like to know if anyone has already managed to use it or if there is a simple boilerplate that performs this function. Could anyone help me with this? Thank you in advance.

**Comment by PHDiniiz** [comment]
I follow all steps (with Redis), but when I send a request on /register, I received this message:

```
"message": "Reached the max retries per request limit (which is 20). Refer to \"maxRetriesPerRequest\" option for details.",
    "stack": "MaxRetriesPerRequestError: Reached the max retries per request limit (which is 20). Refer to \"maxRetriesPerRequest\" option for details.\n    at Socket.<anonymous> (...\node_modules\\ioredis\\built\\redis\\event_handler.js:182:37)\n    at Object.onceWrapper (node:events:629:26)\n    at Socket.emit (node:events:514:28)\n    at TCP.<anonymous> (node:net:323:12)"
```

**Comment by RomainLanz** [comment]
Hey @PHDiniiz! 👋🏻 

I am not sure I got what you mean by "Redis is used instead of Database", but you can use Prisma with [our community provider (adonis-prisma)](https://github.com/wahyubucil/adonis-prisma). However, I would recommend using Lucid and an SQL database instead of relying on MongoDB for your primary database.

Also, the error shown means `ioredis` cannot connect to our Redis instance.

**Reply by PHDiniiz** [reply]
Hii @RomainLanz!

We are currently using version 4 of AdonisJS with Lucid and MongoDB **in production**. However, as AdonisJS is already in its Legacy version, we are migrating to v5 (programming the business rules from scratch). My question is about authentication using Prisma and MongoDB. Using the @adonisjs/auth package, it asks where we want to store the API Tokens (Database/Redis). Is it possible to use Database instead of Redis for this? I ask because, as we already use MongoDB, we would not like, at this moment, to use Redis just for this.

 ![Redis](https://github.com/adonisjs/core/assets/49108363/97e9e331-8119-4e35-a636-70171eaba17c)

**Reply by RomainLanz** [reply]
You have to create a guard that works with MongoDB to use our auth module.
We made [a section in the documentation with an example](https://docs.adonisjs.com/guides/auth/custom-user-provider); could you let me know if that helps?

**Reply by PHDiniiz** [reply]
@RomainLanz 

I'lltest the information in the documentation above and get back to you as soon as possible.
